src/compiler/flatcc.c

```diff
@@ -472,6 +472,11 @@ int flatcc_generate_files(flatcc_context_t ctx)
     }
 #endif
 
+    if (!(P->opts.cgen_reader | P->opts.cgen_builder | P->opts.cgen_verifier
+        | P->opts.cgen_common_reader | P->opts.cgen_common_builder
+        | P->opts.cgen_json_parser | P->opts.cgen_json_printer)) {
+        return 0;
+    }
```

Comment:
I'm not sure this is the place to have this check. I'd rather keep that in the main cli function. fb_init_output does check for gen outfile and fb_endoutput should be neutral if fp is null. But I do not recall all details.